posts/2021-02-15-eclipse-transformer.adoc

```diff
@@ -0,0 +1,311 @@
+open-graph-image: https://openliberty.io/img/twitter_card.jpg
+---
+= Using the Eclipse Transformer for Jakarta EE 9 Development and Test
+Thomas Bitonti <https://github.com/tbitonti>
+
+== Overview
+
+This article describes how application developers can use the Eclipse Transformer to update their application and test artifacts to run using Jakarta EE 9.
+
+The core problem introduced by Jakarta EE 9 is the renaming of package prefixes from "javax" to "jakarta" for API packages and for properties.  As a consequence, existing applications which run against Java EE or Jakarta EE 8 do not run against Jakarta EE 9.  To run existing applications as Jakarta EE 9 applications, references to specific packages and properties must be updated.
+
+To solve this problem, the Open Liberty development team created the Eclipse Transformer.  This open source project provides a utility which updates package references in a variety of artifacts.
+
+The Open Liberty team uses the Eclipse Transformer on existing Java / Jakarta EE implementation archives.  The Open Liberty team also uses the transformer to update test artifacts, which include:
+
+* Test applications;
+* The actual tests;
+* Server configuration files used by the tests;
+
+In a similar fashion, application developers can use the Eclipse Transformer to transform their applications and test artifacts.
+
+The transformer is quick and efficient.  When integrated as a build step, as is done in Open Liberty builds, use of the transformer avoids the creation of new Jakarta EE 9 source files.
+
+== Transformation Strategy
+
+The stratgy used for Jakarta EE 9 test enablement in Open Liberty builds is to annotate test classes and methods to indicate which of these are to be run as Jakarta EE 9 tests.  As an added step, the test framework transforms and re-runs the annotated tests as Jakarta EE 9 tests.
+
+Two specific updates are required when transforming test artifacts.  First, package references are updated, changing Java / Jakarta EE "javax" package references into Jakarta EE 9 "jakarta" package references.  Package references can be in several locations, including test applications, test code, and server configurations.  Second, server configurations are updated to replace Java / Jakarta EE features with corresponding Jakarta EE 9 features.  The Eclipse Transformer is used to perform the first of these updates.  Currently, updates to server configurations must be done manually.  (Server configuration features could be updated using the transformer, if rules data were created with the substitution values needed for this update.)
```

Comment:
For `Currently, updates to server configurations must be done manually. (Server configuration features could be updated using the transformer, if rules data were created with the substitution values needed for this update.)`

The `must` is too strong because you give an alternative that negates the `must` wording.  Maybe something like:

`The Open Liberty test framework updates the feature names in the server configurations using an existing function in the framework.  Alternatively, the transformer could be used to update features in the server configuration if rules data were created with the substitution values needed for this update.`

Reply:
Reworded.